bin/polkadot-archive/src/archive.rs

```diff
@@ -48,45 +48,39 @@ pub fn run_archive<D: ReadOnlyDB + 'static>(config: Config) -> Result<Box<dyn Ar
 
 	match config.cli().chain.to_ascii_lowercase().as_str() {
 		"kusama" | "ksm" => {
-			let archive = ArchiveBuilder::<Block, ksm_rt::RuntimeApi, polkadot_service::KusamaExecutor, D> {
-				pg_url: config.psql_conf().map(|u| u.url()),
-				cache_size: config.cache_size(),
-				block_workers: config.block_workers(),
-				wasm_pages: config.wasm_pages(),
-				max_block_load: config.max_block_load(),
-				..ArchiveBuilder::default()
-			}
-			.chain_data_db(db_path)
-			.chain_spec(spec)
-			.build()?;
+			let archive = ArchiveBuilder::<Block, ksm_rt::RuntimeApi, KusamaExecutor, D>::default()
+				.chain_spec(spec)
+				.chain_data_path(Some(db_path))
+				.pg_url(config.psql_conf().map(|u| u.url()))
+				.cache_size(config.cache_size())
+				.block_workers(config.block_workers())
+				.wasm_pages(config.wasm_pages())
+				.max_block_load(config.max_block_load())
+				.build()?;
 			Ok(Box::new(archive))
 		}
 		"westend" => {
-			let archive = ArchiveBuilder::<Block, westend_rt::RuntimeApi, polkadot_service::WestendExecutor, D> {
-				pg_url: config.psql_conf().map(|u| u.url()),
-				cache_size: config.cache_size(),
-				block_workers: config.block_workers(),
-				wasm_pages: config.wasm_pages(),
-				max_block_load: config.max_block_load(),
-				..ArchiveBuilder::default()
-			}
-			.chain_data_db(db_path)
-			.chain_spec(spec)
-			.build()?;
+			let archive = ArchiveBuilder::<Block, westend_rt::RuntimeApi, WestendExecutor, D>::default()
+				.chain_spec(spec)
+				.chain_data_path(Some(db_path))
+				.pg_url(config.psql_conf().map(|u| u.url()))
+				.cache_size(config.cache_size())
+				.block_workers(config.block_workers())
+				.wasm_pages(config.wasm_pages())
+				.max_block_load(config.max_block_load())
+				.build()?;
 			Ok(Box::new(archive))
 		}
 		"polkadot" | "dot" => {
-			let archive = ArchiveBuilder::<Block, dot_rt::RuntimeApi, polkadot_service::PolkadotExecutor, D> {
-				pg_url: config.psql_conf().map(|u| u.url()),
-				cache_size: config.cache_size(),
-				block_workers: config.block_workers(),
-				wasm_pages: config.wasm_pages(),
-				max_block_load: config.max_block_load(),
-				..ArchiveBuilder::default()
-			}
-			.chain_data_db(db_path)
-			.chain_spec(spec)
-			.build()?;
+			let archive = ArchiveBuilder::<Block, dot_rt::RuntimeApi, PolkadotExecutor, D>::default()
+				.chain_spec(spec)
+				.chain_data_path(Some(db_path))
+				.pg_url(config.psql_conf().map(|u| u.url()))
+				.cache_size(config.cache_size())
+				.block_workers(config.block_workers())
+				.wasm_pages(config.wasm_pages())
+				.max_block_load(config.max_block_load())
+				.build()?;
```

Comment:
Repeating this block of code three times is a bit annoying. It might be a bit of a hassle to improve because of the generics and bounds, but maybe we can do something smart here?

Reply:
yeah, I prefer to change this part in separate PR, especially I plan to optimize the setting part of the configuration file later.